recipes/perl-class-accessor/recipe.yaml

```
@@ -0,0 +1,56 @@
+context:
+  version: 0.51
+
+package:
+  name: perl-class-accessor
+  version: ${{ version }}
+
+source:
+  url: https://cpan.metacpan.org/authors/id/K/KA/KASEI/Class-Accessor-${{ version }}.tar.gz
+  sha256: bf12a3e5de5a2c6e8a447b364f4f5a050bf74624c56e315022ae7992ff2f411c
+
+build:
+  number: 0
+  noarch: generic
+  script:
+    - |
+      perl Makefile.PL INSTALLDIRS=site NO_PERLLOCAL=1 NO_PACKLIST=1
+      make
+      make test
+      make install VERBINST=1
+    - if: unix
+      then: |
+        cp ${PREFIX}/man/man1/perlartistic.1 ${RECIPE_DIR}
+        cp ${PREFIX}/man/man1/perlgpl.1 ${RECIPE_DIR}
+      else: |
+        copy %PREFIX%\man\man1\perlartistic.1 %RECIPE_DIR%
+        copy %PREFIX%\man\man1\perlgpl.1 %RECIPE_DIR%
```

```suggestion
      cp ${PREFIX}/man/man1/perlartistic.1 ${RECIPE_DIR}
      cp ${PREFIX}/man/man1/perlgpl.1 ${RECIPE_DIR}
```
OS selectors like `if: unix` aren't allowed in `noarch: generic` recipes because they are only built on Linux in the feedstock.  For `noarch: generic` recipes, the Windows build failure is simply ignored - hopefully the logic for this will eventually be improved so that the build is actually skipped on Windows entirely.  Sorry if this confusing!

---

Thanks for the continued feedback, @danielnachun!  That arrangement seems to have worked, however the `copy` command failed on windows. Also, strangely, the linter did not point to issues with that `script` arrangement.  Is there explicit failure elsewhere?  Or will it fail later when the feedstock is created?  If so, I'll revert to using the `cp` command only as suggested here.

In a sense, being able to build a `noarch` package in different OSes should be at least allowed, in the grand scheme of things. 

Other than that, @conda-forge/perl-packagers should weigh in here so that we get the linter happy.  Do you know anyone that can do this?

---

> In a sense, being able to build a `noarch` package in different OSes should be at least allowed, in the grand scheme of things.

Unfortunately in practice it's almost never worth the effort to actually get `noarch` packages to build properly on Windows without resorting to platform-specific code. The general idea is that if you are a Windows user and want to build a `noarch` package, you need to do so in a Linux VM.  Windows support in `conda-forge` is essentially optional as it is, so I'd in general rather focus on making nice `noarch` packages that are simple and easy to maintain instead of making things very complicated just to accommodate a niche case of building a `noarch` package on a platform that never gets tested in CI.

---

A little more background here: the biggest issue is that the conda-forge Perl build for Windows currently just re-packages a Windows build called Strawberry Perl, and that build uses different filesystem paths for a bunch of things, so the idea of `noarch` even for pure-perl packages doesn't currently work in practice.

FWIW, about a year ago I spent a lot (too much!) time on [a PR for the Perl feedstock](https://github.com/conda-forge/perl-feedstock/pull/68) that performed a native Windows build of Perl using consistent filesystem paths, which allows for true `noarch` Perl recipes that will pass CI on Windows. The package maintainers never reviewed it and I moved on to other things, but I still hope that something happens with it at some point.

---

I'd love to get a native Windows perl build set up - now that it's easier to communicate with Zulip we should get that going again.  I'd also like to make a bunch of other improvements to our perl infrastructure like adding newer versions, supporting building against multiple versions for native packages, and a perl equivalent to `noarch: python` so we only have to build a Perl package one one version for non-native packages.

---

@danielnachun Sounds good to me...I'm going to try to get some eyes on that PR at last and try to get some movement. About a year ago I began an effort to migrate all of the bioconda perl packages to conda-forge, but it was really hard to get anyone with commit/merge abilities to look at PRs, and I eventually put it on the shelf. If we can get some momentum behind Perl again in conda-forge, maybe I'll pick that back up.

---

I'm working on joining staged-recipes and the Perl migration is one of the main large-scale projects I want to help get done this year.  I have a bunch of Perl packages I want to add myself, and once those get going, we can hopefully continue the momentum to get the migration done.